ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveFilterSortPredicates.java

```diff
@@ -223,7 +224,13 @@ public Double visitCall(RexCall call) {
           return null;
         }
         cost += operandCost;
-        Double size = HiveRelMdSize.averageTypeSize(operand.getType());
```

Comment:
I believe this is the change which have caused lot of operand orders to change
because this seems to be unrelated to HIVE-23100 to me - could it have it's own ticket?

Reply:
Yeah... The reason why I have it in this patch is that a lot of q files were changing too without it. It seems the main reason is the subtle changes in the handling of char, varchar, and string which lead to different ordering.
I could definitely create a new issue for this, but given that it will not reduce the number of q file changes and that the method `typeValueSize` is part of Calcite, could we include it here?